### PR TITLE
feat: Outputs result

### PR DESCRIPTION
**Pull request by kobayu858**
###SUMMARY
- Preparing to output test results
- Creating the file
- Output information on used dag in yaml

### CODE REVIEW

File: lib/src/output.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
+use log::{info, warn};
+use petgraph::Graph;
+use serde_derive::{Deserialize, Serialize};
+use serde_yaml;
+use std::fs;
+
+use crate::graph_extension::{GraphExtension, NodeData};
+
+#[derive(Serialize, Deserialize)]
+struct DAGSet {
```
Comment (atsushi421) [inline lib/src/output.rs:10]:
```suggestion
struct DAGSetInfo {
```

File: lib/src/output.rs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    total_utilization: f32,
+    dag_set: Vec<DAG>,
+}
+
+#[derive(Serialize, Deserialize)]
+#[allow(clippy::upper_case_acronyms)]
+struct DAG {
```
Comment (atsushi421) [inline lib/src/output.rs:17]:
```suggestion
struct DAGInfo {
```

File: lib/src/output.rs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    let file_path = format!("{}/{}.yaml", folder_path, file_name);
+    if let Err(err) = fs::File::create(&file_path) {
+        warn!("Failed to create file: {}", err);
+    }
+    file_path
+}
+
+pub fn dag_set_info_to_yaml(mut dag_set: Vec<Graph<NodeData, f32>>, file_path: &str) {
+    let mut total_utilization = 0.0;
+    let mut dag_instances = Vec::new();
```
Comment (atsushi421) [inline lib/src/output.rs:37]:
```suggestion
    let mut each_dag_info = Vec::new();
```

File: lib/src/output.rs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    for dag in dag_set.iter_mut() {
+        let volume = dag.get_volume();
+        let period = dag.get_head_period().unwrap();
+        let critical_path = dag.get_critical_path();
+        let critical_path_length = dag.get_total_wcet_from_nodes(&critical_path);
+        total_utilization += volume / period;
+
+        let dag_instance = DAG {
```
Comment (atsushi421) [inline lib/src/output.rs:46]:
```suggestion
        let dag_info = DAGInfo {
```

File: lib/src/output.rs [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            critical_path_length,
+            end_to_end_deadline: period,
+            volume,
+        };
+
+        dag_instances.push(dag_instance);
+    }
+
+    let dag_set = DAGSet {
```
Comment (atsushi421) [inline lib/src/output.rs:55]:
```suggestion
    let dag_set_info = DAGSetInfo {
```

**Review by atsushi421 — CHANGES_REQUESTED** [review]

File: lib/src/output.rs (hunk 1 above)
Comment (kobayu858) [reply lib/src/output.rs]: accept your suggestion.

File: lib/src/output.rs (hunk 2 above)
Comment (kobayu858) [reply lib/src/output.rs]: accept your suggestion.

File: lib/src/output.rs (hunk 3 above)
Comment (kobayu858) [reply lib/src/output.rs]: accept your suggestion.

File: lib/src/output.rs (hunk 4 above)
Comment (kobayu858) [reply lib/src/output.rs]: accept your suggestion.

File: lib/src/output.rs (hunk 5 above)
Comment (kobayu858) [reply lib/src/output.rs]: accept your suggestion.

File: lib/src/output_log.rs [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+use serde_derive::{Deserialize, Serialize};
+use serde_yaml;
+use std::fs;
+
+use crate::graph_extension::{GraphExtension, NodeData};
+
+#[derive(Serialize, Deserialize)]
+struct DAGSetInfo {
+    total_utilization: f32,
+    dag_set: Vec<DAGInfo>,
```
Comment (atsushi421) [inline lib/src/output_log.rs:12]:
```suggestion
    eadh_dag_info: Vec<DAGInfo>,
```

**Review by atsushi421 — CHANGES_REQUESTED** [review]

File: lib/src/output_log.rs (hunk 6 above)
Comment (kobayu858) [reply lib/src/output_log.rs]: accept your suggestion.

**Review by atsushi421 — APPROVED** [review]
LGTM！

**Pull request merged**